core/base.py

```diff
@@ -0,0 +1,323 @@
+            'Cannot use_external_resource because resource {0} '
+            'is not in this account.'.format(resource_id))
+
+
+class AwsBaseRelationship(AwsBase):
+
+    def __init__(self, client=None):
+        super(AwsBaseRelationship, self).__init__(client)
+        self.source_resource_id = \
+            ctx.source.instance.runtime_properties.get(
+                constants.EXTERNAL_RESOURCE_ID, None) if \
+            constants.EXTERNAL_RESOURCE_ID in \
+            ctx.source.instance.runtime_properties.keys() else \
+            ctx.source.node.properties['resource_id']
+        self.target_resource_id = ctx.target.instance.runtime_properties.get(
+                constants.EXTERNAL_RESOURCE_ID, None) if \
+            constants.EXTERNAL_RESOURCE_ID in \
+            ctx.target.instance.runtime_properties.keys() else \
+            ctx.target.node.properties['resource_id']
+        self.source_external_resource = ctx.source.node.properties['use_external_resource']
+        self.target_external_resource = ctx.target.node.properties['use_external_resource']
+        self.source_get_all_matching = None
+        self.target_get_all_matching = None
+        self.source_get_all_argument = None
+        self.target_get_all_argument = None
+
+    def associate(self):
+        return False
+
+    def associated(self):
+        if self.associate_external_resource():
+            return self.post_associate()
+        elif self.associate():
+            return self.post_associate()
+
+        raise NonRecoverableError('Unable to associate {0} with {1}.'.format(self.source_resource_id, self.target_resource_id))
+
+    def associate_external_resource(self):
+
+        if not self.source_external_resource:
+            return False
+
+        resource = self.source_get_all_matching([self.source_resource_id])[0]
+
+        if resource is None:
+            self.raise_cannot_use_external_resource(self.source_resource_id)
+
+        ctx.logger.info('not associating external resource {0}.'.format(resource.id))
+
+        return True
+
+    def disassociate(self):
+        return False
+
+    def disassociated(self):
+        if self.disassociate_external_resource():
+            return self.post_disassociate()
+        elif self.disassociate():
+            return self.post_disassociate()
+        raise NonRecoverableError('Unable to disassociate {0} from {1}.'.format(self.source_resource_id, self.target_resource_id))
+
+    def disassociate_external_resource(self):
+        if not self.source_external_resource:
+            return False
+        resource = self.source_get_all_matching([self.source_resource_id])[0]
+        ctx.logger.info('not disassociating external resource {0}.'.format(resource.id))
+        return True
+
+    def post_associate(self):
+        return True
+
+    def post_disassociate(self):
+        return True
+
+class AwsBaseNode(AwsBase):
+
+    def __init__(self,
+                 aws_resource_type,
```

Comment:
inside of constants
dhcp = {
  aws: kjdsfldsj,
  cloudfy: lkjsdf
}
aws_resource_type = constants.dhcp.aws
cloudify_resource_type = constants.dhcp.cloudify